From 5b9499de5cb9ae86ea766af6a77450f739499bd3 Mon Sep 17 00:00:00 2001
From: "Gao, HuadongX" <huadongx.gao@intel.com>
Date: Tue, 21 Nov 2023 10:24:05 +0000
Subject: [PATCH] Add buffered for PrintWriter in dumpGfxInfo

The monkey test error:
java.lang.RuntimeException: StrictMode ThreadPolicy violation
...
Caused by: android.os.strictmode.UnbufferedIoViolation

The resolve method:
PrintWrite use BufferedWriter instead of FastPrintWriter.

Tracked-On: OAM-113172
Signed-off-by: Gao, HuadongX <huadongx.gao@intel.com>
---
 core/java/android/view/WindowManagerGlobal.java | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/core/java/android/view/WindowManagerGlobal.java b/core/java/android/view/WindowManagerGlobal.java
index c92a3a086a8b..3f57a3dc71ec 100644
--- a/core/java/android/view/WindowManagerGlobal.java
+++ b/core/java/android/view/WindowManagerGlobal.java
@@ -40,6 +40,8 @@ import com.android.internal.util.FastPrintWriter;
 import java.io.FileDescriptor;
 import java.io.FileOutputStream;
 import java.io.PrintWriter;
+import java.io.BufferedWriter;
+import java.io.OutputStreamWriter;
 import java.util.ArrayList;
 
 /**
@@ -598,9 +600,9 @@ public final class WindowManagerGlobal {
         }
     }
 
-    public void dumpGfxInfo(FileDescriptor fd, String[] args) {
+    public void dumpGfxInfo(FileDescriptor fd, String[] args) throws Exception {
         FileOutputStream fout = new FileOutputStream(fd);
-        PrintWriter pw = new FastPrintWriter(fout);
+        PrintWriter pw = new PrintWriter(new BufferedWriter(new OutputStreamWriter(fout)));
         try {
             synchronized (mLock) {
                 final int count = mViews.size();
-- 
2.34.1

